src/util/misc/projectStroke.ts

```diff
@@ -39,62 +49,216 @@ export const projectStrokeOnPoints = (
     strokeUniform,
     scaleX,
     scaleY,
+    skewX,
+    skewY
   }: projectStrokeOnPointsOptions,
   openPath: boolean
-): Point[] => {
-  const coords: Point[] = [],
+): {projectedPoint: Point, originPoint: Point, bisector?: { vector: Point, angle: number }}[] => {
+
+  const coords: {projectedPoint: Point, originPoint: Point, bisector?: { vector: Point, angle: number }}[] = [],
     s = strokeWidth / 2,
+    scale = new Point(scaleX, scaleY),
     strokeUniformScalar = strokeUniform
       ? new Point(1 / scaleX, 1 / scaleY)
       : new Point(1, 1),
-    getStrokeHatVector = (v: Point) => {
-      const scalar = s / Math.hypot(v.x, v.y);
-      return new Point(
-        v.x * scalar * strokeUniformScalar.x,
-        v.y * scalar * strokeUniformScalar.y
-      );
+    scaleHatVector = (hatVector: Point, scalar: number) => {
+      return hatVector.multiply(strokeUniformScalar).scalarMultiply(scalar);
+    },
+    applySkew = (v: Point) => {
+      let vector = new Point(v);
+      vector.y += vector.x * Math.tan(degreesToRadians(skewY)) // skewY must be applied before skewX as this distortion affects skewX calculation
+      vector.x += vector.y * Math.tan(degreesToRadians(skewX))
+      return vector
     };
+
   if (points.length <= 1) {
     return coords;
   }
+
   points.forEach(function (p, index) {
-    const A = new Point(p.x, p.y);
-    let B, C;
+    let A = new Point(p.x, p.y), B, C;
     if (index === 0) {
       C = points[index + 1];
-      B = openPath
-        ? getStrokeHatVector(createVector(C, A)).add(A)
-        : points[points.length - 1];
-    } else if (index === points.length - 1) {
+      B = openPath ? A : points[points.length - 1];
+    }
+    else if (index === points.length - 1) {
       B = points[index - 1];
-      C = openPath ? getStrokeHatVector(createVector(B, A)).add(A) : points[0];
-    } else {
+      C = openPath ? A : points[0];
+    }
+    else {
       B = points[index - 1];
       C = points[index + 1];
     }
-    const bisector = getBisector(A, B, C),
+
+    //  safeguard in case `points` are not `Point`
+    B = new Point(B.x, B.y);
+    C = new Point(C.x, C.y);
+
+    /* OPEN PATH
+      If open path, no matter the type of the line join, the line is always the same
+      Calculation: to find the projections, just find the points orthogonal to that stroke 
+      Visually detailed here: https://github.com/fabricjs/fabric.js/issues/8025
+      TODO: take into account line-cap (I believe that should change the projections)
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/lineCap#comparison_of_line_caps

for sure

![image](https://user-images.githubusercontent.com/34343793/193514744-38f2ecdf-3208-43d1-86f1-fdcea6db98be.png)